vpc/service/validator.go

```diff
@@ -19,9 +21,94 @@ func (vpcService *vpcService) ValidateAllocationParameters(ctx context.Context,
 	defer span.End()
 	log := ctxlogrus.Extract(ctx)
 	ctx = logger.WithLogger(ctx, log)
-	_ = ctx
+	span.AddAttributes(
+		trace.StringAttribute("account", req.AccountId),
+		trace.StringAttribute("subnet-ids", fmt.Sprintf("%v", req.Subnets)),
+		trace.StringAttribute("security-groups", fmt.Sprintf("%v", req.Subnets)),
+	)
+	tx, err := vpcService.db.BeginTx(ctx, &sql.TxOptions{
+		ReadOnly: true,
+	})
+	if err != nil {
+		err = tracehelpers.WithGRPCStatusCode(fmt.Errorf("Could not start database transaction: %w", err), codes.Unknown)
+		tracehelpers.SetStatus(err, span)
+		return nil, err
+	}
 
-	err := status.Error(codes.Unimplemented, "Call not yet implemented")
-	tracehelpers.SetStatus(err, span)
-	return nil, err
+	defer func() {
+		_ = tx.Rollback()
+	}()
+
+	var validationFailures []*titus.ParametersValidationResponse_ValidationFailure
+
+	row := tx.QueryRow("SELECT count(*) FROM accounts WHERE account_id = $1", req.AccountId)
+	var accountsFound int
+	err = row.Scan(&accountsFound)
+	if err != nil {
+		err = fmt.Errorf("Could not scan accounts found count: %w", err)
+		tracehelpers.SetStatus(err, span)
+		return nil, err
+	}
+	if accountsFound == 0 {
+		validationFailures = append(validationFailures, &titus.ParametersValidationResponse_ValidationFailure{
+			FailureOneOf: &titus.ParametersValidationResponse_ValidationFailure_AccountIdUnsupported{
+				AccountIdUnsupported: &titus.ParametersValidationResponse_AccountIdUnsupported{
+					AccountId: req.AccountId,
+				},
+			},
+		})
+	}
+
+	rows, err := tx.QueryContext(ctx, `
```

Comment:
Check `req.Subnets` has one or more values to see if this validation step is required?

Reply:
If subnets is empty doesn't that fail validation?